opentelemetry-http/Cargo.toml

```diff
@@ -9,6 +9,9 @@ license = "Apache-2.0"
 edition = "2021"
 rust-version = "1.65"
 
+[features]
+reqwest-rustls = ["reqwest", "reqwest/rustls-tls"]
```

Comment:
Just curious, in other parts of [code](https://github.com/open-telemetry/opentelemetry-rust/blob/815e3874a28f132415e7eead92385ff72ef2c9a2/opentelemetry-zipkin/Cargo.toml#L26), the feature depends on the `rustls-tls-native-roots`, here it is `rustls-tls`. The difference is the choice of the root-certificates crate. Does it makes any difference ?

Reply:
Thanks @LuisOsta. I went through that discussion before commenting. The concern I had is - if the application is running in an environment that relies on enterprise-specific root certificates (such as those used for a mandatory proxy or internal CA), `webpki-roots` will not include these out of the box. The `rustls-tls-native-roots` looks more flexible in that regard. But I am not an expert here, so will leave it for others to comment.

Reply:
@lalitb That's very fair, honestly seems like both have their tradeoffs. I'll switch to `rustls-tls-native-roots` to remain consistent then